method_formatter.go

```diff
@@ -0,0 +1,26 @@
+package jsonrpc
+
+import "strings"
+
+// MethodNameFormatter is a function that takes a namespace and a method name and returns the full method name, sent via JSON-RPC.
+// This is useful if you want to customize the default behaviour, e.g. send without the namespace or make it lowercase.
+type MethodNameFormatter func(namespace, method string) string
+
+// DefaultMethodNameFormatter joins the namespace and method name with a dot.
+func DefaultMethodNameFormatter(namespace, method string) string {
+	return namespace + "." + method
+}
+
+// NoNamespaceMethodNameFormatter returns the method name as is, without the namespace.
+func NoNamespaceMethodNameFormatter(_, method string) string {
+	return method
+}
+
+// NoNamespaceDecapitalizedMethodNameFormatter returns the method name as is, without the namespace, and decapitalizes the first letter.
+// e.g. "Inc" -> "inc"
+func NoNamespaceDecapitalizedMethodNameFormatter(_, method string) string {
+	if len(method) == 0 {
+		return ""
+	}
+	return strings.ToLower(method[:1]) + method[1:]
+}
```

Comment:
```suggestion
// CaseStyle represents the case style for method names.
type CaseStyle int

const (
    OriginalCase CaseStyle = iota
    LowerFirstCharCase
)

// NewMethodNameFormatter creates a new method name formatter based on the provided options.
func NewMethodNameFormatter(includeNamespace bool, nameCase  CaseStyle) MethodNameFormatter {
    return func(namespace, method string) string {
        formattedMethod := method
        if nameCase == LowerFirstCharCase && len(method) > 0 {
            formattedMethod = strings.ToLower(method[:1]) + method[1:]
        }
        if includeNamespace {
            return namespace + "." + formattedMethod
        }
        return formattedMethod
    }
}

// DefaultMethodNameFormatter is a pass-through formatter with default options.
var DefaultMethodNameFormatter = NewMethodNameFormatter(true, OriginalCase)
```

How about something like this to solve the weird naming problem?

Reply:
Good idea 💯

Reply:
✅ Done